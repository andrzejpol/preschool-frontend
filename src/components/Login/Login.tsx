import React, {useState} from 'react';
import Logo from "../../assets/logo.svg";
import {Link, useNavigate} from "react-router-dom";
import {login} from "../../../authService.ts"
import {useDispatch} from "react-redux";
import {hideGlobalLoader, showGlobalLoader} from "../../slices/loaderSlice.ts";

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const handleLogin = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    dispatch(showGlobalLoader());
    try {
      const response = await login(email, password);
      console.log(response);
      navigate("/dashboard");
    } catch (error) {
      console.log("Błąd logowania", error);
    } finally {
      dispatch(hideGlobalLoader());
    }
  };

  return (
    <div
      className="flex items-center justify-center min-h-screen bg-gradient-to-b from-preschoolPrimary to-preschoolPrimaryLight">
      <form onSubmit={handleLogin} className="flex flex-col gap-5 items-center bg-white rounded-md p-10">
        <img src={Logo} alt="logo" width={150} height={150}/>
        <div className="flex flex-col gap-2">
          <label htmlFor="email" className="text-xs text-gray-500">Email:</label>
          <input
            type="text"
            id="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            required
            className="border border-solid border-gray-500 rounded-[5px] p-1 text-gray-500 focus:border-preschoolPrimary focus:bg-gray-100 focus:outline-none"
          />
        </div>
        <div className="flex flex-col gap-2">
          <label htmlFor="password" className="text-xs text-gray-500">Password:</label>
          <input
            type="password"
            id="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            required
            className="border border-solid border-gray-500 rounded-[5px] p-1 text-gray-500 focus:border-preschoolPrimary focus:bg-gray-100 focus:outline-none"
          />
        </div>
        <button type="submit"
                className="py-2 px-5 text-gray-500 border border-solid border-preschoolPrimary rounded-[5px] hover:bg-preschoolPrimary hover:text-white">Log
          In
        </button>
        <div className="flex gap-2">
          <span className="text-xs">Don't have an account?</span>
          <Link to="/register" className="text-xs text-preschoolPrimary cursor-pointer">Register me</Link>
        </div>
      </form>
    </div>
  );
};

export default Login;
