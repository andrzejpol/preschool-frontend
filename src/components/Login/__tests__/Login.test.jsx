import React from "react";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter as Router} from "react-router-dom";
import configureStore from "redux-mock-store";
import Login from "../Login";

const mockStore = configureStore([]);

describe("Login Component", () => {
    let store;

    beforeEach(() => {
        store = mockStore({});
    });

    it("matches the snapshot", () => {
        const {asFragment} = render(
            <Provider store={store}>
                <Router>
                    <Login/>
                </Router>
            </Provider>
        );

        // Snapshot assertion
        expect(asFragment()).toMatchSnapshot();
    });
});
