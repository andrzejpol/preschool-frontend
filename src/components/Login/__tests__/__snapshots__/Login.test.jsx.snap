// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center min-h-screen bg-gradient-to-b from-preschoolPrimary to-preschoolPrimaryLight"
  >
    <form
      class="flex flex-col gap-5 items-center bg-white rounded-md p-10"
    >
      <img
        alt="logo"
        height="150"
        src="/src/assets/logo.svg"
        width="150"
      />
      <div
        class="flex flex-col gap-2"
      >
        <label
          class="text-xs text-gray-500"
          for="email"
        >
          Email:
        </label>
        <input
          class="border border-solid border-gray-500 rounded-[5px] p-1 text-gray-500 focus:border-preschoolPrimary focus:bg-gray-100 focus:outline-none"
          id="email"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="flex flex-col gap-2"
      >
        <label
          class="text-xs text-gray-500"
          for="password"
        >
          Password:
        </label>
        <input
          class="border border-solid border-gray-500 rounded-[5px] p-1 text-gray-500 focus:border-preschoolPrimary focus:bg-gray-100 focus:outline-none"
          id="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="py-2 px-5 text-gray-500 border border-solid border-preschoolPrimary rounded-[5px] hover:bg-preschoolPrimary hover:text-white"
        type="submit"
      >
        Log In
      </button>
      <div
        class="flex gap-2"
      >
        <span
          class="text-xs"
        >
          Don't have an account?
        </span>
        <a
          class="text-xs text-preschoolPrimary cursor-pointer"
          data-discover="true"
          href="/register"
        >
          Register me
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
